tests/StreamTest.php

```diff
@@ -66,6 +66,15 @@ public function testChecksEof()
         $stream->close();
     }
 
+    public function testReadError()
+    {
+        $this->expectException(\RuntimeException::class);
```

Comment:
Use the following assertion to make sure you are testing the `RuntimeException` that is being introduced by this PR:
```suggestion
        $this->expectException(\RuntimeException::class);
        $this->expectExceptionMessage('Unable to read from stream');
```
For me this sadly breaks the test 😞 So doing a `->read(4)` from an append-only handle is not enough to trigger the `false` value.

Reply:
I have no idea so that fread return false :disappointed:, just remove this test.

Reply:
Thanks for the update!